Sources/StreamChat/Utils/QueuedExecutor.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+private struct QueuedExecutorIdentifier {
+    static var identifier = 0
+}
+
+class QueuedExecutor {
+    typealias Completion = (Error?) -> Void
+    typealias ExecutorClosure = (_: @escaping Completion) -> Void
```

Comment:
I don't get why we need this. The ticket clearly says the implementation should be the same as the channel and message pagination. I would recommend taking a look at `ChannelController` and use the same strategy to do pagination so that the code is consistent across the codebase.

Reply:
I will check for sure, 

This approach handles multiple requests on same range and also concurrent requests for different ranges, 
with tests this should be more clear

Reply:
In this case all the requests are put to serial queue which is executing them 1-by-1 and never double-loads any ranges, which can be caused by incorrect UI implementation.